### ISSUE TITLE
Bug: Adding name delimiter introduced formatting bug.

### ISSUE THREAD

**Issue by kevintechie** [fullstack-build/tslog#174]
**Describe the bug**
I'm not sure of how the new delimiter setting is supposed to be used, but as implemented, it causes extra white space (tab) to be added to the end of a logger name if it exists. Perhaps, the template string replacement is not the correct location for adding the delimiter. Either that or the delimiter should be blank "" as the default.

Another possibility is to add another placeholder (nameWithDelimiter) for adding of the delimiter string when wanted.

**To Reproduce**

```
test("prettyLogTemplate - name placeholder", (): void => {
  const logger = new Logger({
    type: "pretty",
    name: "asdf",
    prettyLogTemplate: "**{{name}}**",
    stylePrettyLogs: false,
  });
  logger.log(1234, "testLevel", "Test");
  expect(getConsoleLog()).toContain(`**asdf**`);
  expect(getConsoleLog()).toContain("** Test");
});
```
**Expected behavior**
Name placeholder should not add any additional formatting.

**Node.js Version**
16

**OS incl. Version**
Ubuntu 20.04

**Comment by terehov** [comment]
I believe the default behaviour should work the best out of the box. 
I decided to introduce two new template variables `nameWithDelimiterPrefix`and `nameWithDelimiterSuffix`.

**Comment by kevintechie** [comment]
I like the approach of returning name without delimiter and adding the two new delimiter template variables. This way it unambiguous about what you'll see.

**Comment by terehov** [comment]
Fixed in `V4.3.0`. Just released.
